numpy/distutils/ccompiler.py

```diff
@@ -620,7 +640,19 @@ def matcher(version_string):
             version = m.group('version')
             return version
 
-    status, output = exec_command(version_cmd, use_tee=0)
+    try:
+        output = subprocess.check_output(version_cmd)
+    except subprocess.CalledProcessError as exc:
+        output = exc.output
+        status = exc.returncode
+    except OSError:
+        # match the historical returns for a parent
+        # exception class caught by exec_command()
+        status = 127
+        output = ''
+    else:
+        output = filepath_from_subprocess_output(output)
```

Comment:
I think we're expecting a version string here and Eric has previously criticized extraneous usage of our new `filepath_from_subprocess_output` -- it may be possible to remove this, but note that `exec_command` was not able to make the distinction between filepaths and version strings so it would also perform the processing steps / control flow invoked by this function anyway.